arch/x86/core/acpi.c

```diff
@@ -5,6 +5,10 @@
 #include <kernel.h>
 #include <arch/x86/acpi.h>
 
+#define RSDP_UNSEARCHED (void *)0xFFFFFFFF
```

Comment:
I am not super confident about this trick (what if the pointer is above 4G, the first 32bits could be made of that value perhaps?). Idea was to save one bool is_found; but ... well it's on x86 with quite a good amount of memory so maybe we don't care.

Reply:
IIRC the spec requires that ACPI table address be aligned to 16 bytes (or 8?  I think it's 16).  If you need a value that is guaranteed illegal then anything with some low bits set is likely fine.  This looks good to me.

Reply:
FWIW I doubt we will ever care about being thrifty with memory on a PC-like system. We have to skip most of the first megabyte of RAM anyway. Keep this in mind when considering any memory optimizations that might not be intuitive. Consider just using NULL, we *never* map the NULL virtual address.

side note: eventually all this legacy stuff needs to go. We really just need to get the RSDP out of UEFI. This kind of legacy probing is not guaranteed to work on a UEFI system. I opened some bugs about this, no need to address this PR

Reply:
idea here is to lookup for rsdp only once. So NULL is not the right value to determine if we already looked for it. I'll just have a boolean.